neo/SmartContract/ApplicationEngine.cs

```diff
@@ -503,91 +503,42 @@ protected virtual long GetPriceForSysCall()
             byte length = CurrentContext.Script[CurrentContext.InstructionPointer + 1];
             if (CurrentContext.InstructionPointer > CurrentContext.Script.Length - length - 2)
                 return 1;
-            string api_name = Encoding.ASCII.GetString(CurrentContext.Script, CurrentContext.InstructionPointer + 2, length);
-            switch (api_name)
+            uint api_hash = length == 4
+                ? System.BitConverter.ToUInt32(CurrentContext.Script, CurrentContext.InstructionPointer + 2)
+                : Encoding.ASCII.GetString(CurrentContext.Script, CurrentContext.InstructionPointer + 2, length).ToInteropMethodHash();
+            long price = Service.GetPrice(api_hash);
+            if (price > 0) return price;
+            if (api_hash == "Neo.Asset.Create".ToInteropMethodHash() ||
```

Comment:
We should hardcode the values, for speed up

Reply:
I agree it could be better, but now it's just a dictionary lookup, embedded in ToInteropMethodHash helper: https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/Helper.cs#L11